crates/vm/levm/src/memory.rs

```diff
@@ -117,9 +118,9 @@ impl Memory {
         // SAFETY: resize already makes sure bounds are correct.
         #[allow(unsafe_code)]
         unsafe {
-            Ok(buf
-                .get_unchecked(true_offset..(true_offset.wrapping_add(size)))
-                .to_vec())
+            Ok(Bytes::copy_from_slice(buf.get_unchecked(
+                true_offset..(true_offset.wrapping_add(size)),
+            )))
```

Comment:
If we're doing this we should probably just make sure `Memory` is of type `Bytes` and use the `slice` method, as this returns a reference counted slice akin to those from Go (in their case GC'd but still a pair of pointers to the old array).
Also, keep in mind `Bytes` just took ownership from the old `Vec` and converted it to a reference counted thingy, so it should be the same number of allocations with this change.